controller/src/beerocks/master/tasks/client_steering_task.cpp

```diff
@@ -148,24 +172,47 @@ void client_steering_task::steer_sta()
         /*
         * send disallow to all others
         */
-        Socket *sd   = database.get_node_socket(hostap);
-        auto request = message_com::create_vs_message<
-            beerocks_message::cACTION_CONTROL_CLIENT_DISALLOW_REQUEST>(cmdu_tx, id);
+        Socket *sd = database.get_node_socket(hostap);
+        if (!cmdu_tx.create(0,
+                            ieee1905_1::eMessageType::CLIENT_ASSOCIATION_CONTROL_REQUEST_MESSAGE)) {
+            LOG(ERROR)
+                << "cmdu creation of type CLIENT_ASSOCIATION_CONTROL_REQUEST_MESSAGE, has failed";
+            return;
+        }
 
-        if (request == nullptr) {
-            LOG(ERROR) << "Failed building ACTION_CONTROL_CLIENT_DISALLOW_REQUEST message!";
+        auto association_control_block_request_tlv =
+            cmdu_tx.addClass<wfa_map::tlvClientAssociationControlRequest>();
+        if (!association_control_block_request_tlv) {
+            LOG(ERROR) << "addClass wfa_map::tlvClientAssociationControlRequest failed";
             return;
         }
-        request->mac() = network_utils::mac_from_string(sta_mac);
+
+        association_control_block_request_tlv->bssid_to_block_client() =
+            network_utils::mac_from_string(current_ap_mac);
+        association_control_block_request_tlv->association_control() =
+            wfa_map::tlvClientAssociationControlRequest::BLOCK;
+        association_control_block_request_tlv->validity_period_sec() = disassoc_timer;
```

Comment:
I don't think this is correct... First of all, of course, disassoc_timer is expressed in TU while validity period is expressed in seconds, so that's a factor 1000 different. But also, the block should definitely still be in place immediately after the disassociation took place. So a more reasonable validity period, I think, would be the timeout used by the client steering task, i.e. steering_wait_time_ms/1000. In fact, that would remove the need to do unblocking in the FINALIZE state.

But again, that's for a different day. Could you make an issue for it?

Reply:
Yes 👍 @arnout

Reply:
#432